packages/block-editor/src/components/publish-date-time-picker/index.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	DateTimePicker,
+	__experimentalHStack as HStack,
+	__experimentalSpacer as Spacer,
+	Button,
+} from '@wordpress/components';
+import { closeSmall } from '@wordpress/icons';
+import { __ } from '@wordpress/i18n';
+import { forwardRef } from '@wordpress/element';
+
+function PublishDateTimePicker(
+	{ onClose, onChange, ...additionalProps },
+	ref
+) {
+	return (
+		<div ref={ ref } className="block-editor-publish-date-time-picker">
+			{ /* TODO: This header is essentially the same as the one in <PostVisiblity />. DRY it up. */ }
+			<HStack className="block-editor-publish-date-time-picker__header">
+				<h2 className="block-editor-publish-date-time-picker__heading">
+					{ __( 'Publish' ) }
+				</h2>
+				<Spacer />
+				<Button
+					className="block-editor-publish-date-time-picker__reset"
+					variant="tertiary"
+					onClick={ () => onChange?.( null ) }
```

Comment:
Is there a reason why `onChange` is treated as optional but `onClose` not?

No biggie, just asking.

Reply:
From my reading of it, it looks like `onClose` is passed as a reference to the Button's `onClick`, so it can happily be `undefined`, whereas in this handler, `onChange` is being called, so it needs optional chaining in case it's `undefined`? I could be wrong, though!

Reply:
With `onChange` I'm just matching the behaviour of `DateTimePicker` which has it as optional.

You're right that `onClose` should probably also be optional for consistency. 

It just so happens that the next PR I have lined up (https://github.com/WordPress/gutenberg/pull/41362) makes `onClose` optional so let's avoid re-triggering GitHub Actions yet again and wait for that.

Reply:
Thanks for the explanation!